Makefile

```diff
@@ -70,11 +69,12 @@ test: ## Run tests (used by Concourse)
 test-with-docker: ## Run tests in Docker container
 	./scripts/run_with_docker.sh make test
 
-.PHONY: upload-to-dockerhub
-upload-to-dockerhub:
+.PHONY: upload-to-docker-registry
+upload-to-docker-registry: ## Upload the current version of the docker image to Docker registry
 	docker build -f docker/Dockerfile -t ${DOCKER_IMAGE_NAME} .
-	$(if ${DOCKERHUB_PASSWORD},,$(error Must specify DOCKERHUB_PASSWORD))
-	@docker login -u govuknotify -p ${DOCKERHUB_PASSWORD}
+	$(if ${DOCKER_USER_NAME},,$(error Must specify DOCKER_USER_NAME))
+	$(if ${CF_DOCKER_PASSWORD},,$(error Must specify CF_DOCKER_PASSWORD))
+	@docker login ${DOCKER_IMAGE} -u ${DOCKER_USER_NAME} -p ${CF_DOCKER_PASSWORD}
```

Comment:
CF_DOCKER_PASSWORD implies this is something to do with cloud foundry? is this just meant to be the container registry password?

Reply:
Yup, that's right. It's used implicitly by `cf push` and I wanted to avoid having a separate `DOCKER_PASSWORD` variable (and check) set to the same thing. I appreciate it's a bit confusing, though.

Do you think we should just stick with two variables?

Reply:
Ah I see, I didn't realise that it's used by CF push to allow it to grab the image from the registry! That makes sense. Looks good as it is!